### ISSUE TITLE
Teslamate stopped working after 2023.38.4

### ISSUE THREAD

**Issue by brontide** [adriankumpf/teslamate#3381]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

After updating to 2023.32.8 this morning the logs are showing the following errors. I have tried the dockerhub tags latest and edge as well as building the current master branch of this repository without change in behavior.

### Relevant log output

```shell
teslamate_1  | 2023-10-27 00:51:16.661 car_id=1 [info] Start / :online
teslamate_1  | 2023-10-27 00:51:16.666 [error] GenStateMachine :"1" terminating
teslamate_1  | ** (MatchError) no match of right hand side value: {:error, #Ecto.Changeset<action: :insert, changes: %{date: ~U[2023-10-27 00:51:16.569000Z], power: 0, odometer: 5419.046258, tpms_pressure_fl: Decimal.new("3.0"), tpms_pressure_fr: Decimal.new("2.975"), tpms_pressure_rl: Decimal.new("3.0"), tpms_pressure_rr: Decimal.new("2.95"), battery_heater: false, driver_temp_setting: Decimal.new("22.0"), fan_status: 0, inside_temp: Decimal.new("30.3"), is_climate_on: false, is_front_defroster_on: false, is_rear_defroster_on: false, outside_temp: Decimal.new("30.0"), passenger_temp_setting: Decimal.new("22.0"), battery_heater_on: false, battery_level: 45, usable_battery_level: 45, ideal_battery_range_km: Decimal.new("240.61"), rated_battery_range_km: Decimal.new("240.61"), est_battery_range_km: Decimal.new("261.47")}, errors: [latitude: {"can't be blank", [validation: :required]}, longitude: {"can't be blank", [validation: :required]}], data: #TeslaMate.Log.Position<>, valid?: false>}
teslamate_1  |     (teslamate 1.28.0-dev) lib/teslamate/vehicles/vehicle.ex:721: anonymous fn/4 in TeslaMate.Vehicles.Vehicle.handle_event/4
teslamate_1  |     (ecto_sql 3.10.1) lib/ecto/adapters/sql.ex:1203: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4
teslamate_1  |     (db_connection 2.5.0) lib/db_connection.ex:1630: DBConnection.run_transaction/4
teslamate_1  |     (teslamate 1.28.0-dev) lib/teslamate/vehicles/vehicle.ex:713: TeslaMate.Vehicles.Vehicle.handle_event/4
teslamate_1  |     (stdlib 5.1.1) gen_statem.erl:1379: :gen_statem.loop_state_callback/11
teslamate_1  |     (stdlib 5.1.1) proc_lib.erl:241: :proc_lib.init_p_do_apply/3
```

### Type of installation

Docker

### Version

latest, edge, and a direct build of mater.

**Comment by brontide** [comment]
I did look but upon further review this _may_ be a duplicate of #3371 but I can't be 100% sure.  I have tried to build and run the master branch with no change in behavior.

**Comment by brontide** [comment]
Tried #3371 patch and now am getting a http 408 errror

```
teslamate_1  | 2023-10-27 01:57:02.059 car_id=1 [info] Connecting ...
teslamate_1  | 2023-10-27 01:57:41.486 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/XXXXXXXXXX/vehicle_data?endpoints=location_data -> 408 (9137.505 ms)
teslamate_1  | 2023-10-27 01:57:41.486 [error] Task #PID<0.2690.0> started from :"1" terminating
teslamate_1  | ** (KeyError) key :drive_state not found in: %TeslaApi.Error{
teslamate_1  |   reason: :unknown,
teslamate_1  |   message: "%{\"error\" => \"{\\\"error\\\": \\\"timeout\\\"}\", \"error_description\" => \"\", \"response\" => nil}",
teslamate_1  |   env: %Tesla.Env{
teslamate_1  |     method: :get,
teslamate_1  |     url: "https://owner-api.teslamotors.com/api/1/vehicles/3744612307359505/vehicle_data?endpoints=location_data",
teslamate_1  |     query: [],
teslamate_1  |     headers: [
teslamate_1  |       {"x-xss-protection", "1; mode=block"},
teslamate_1  |       {"x-txid", "stripped"},
teslamate_1  |       {"content-type", "application/json; charset=utf-8"},
teslamate_1  |       {"x-request-id", "40f6bff2-2bf3-4df1-a183-5b59a5fc7072"},
teslamate_1  |       {"content-length", "75"},
teslamate_1  |       {"x-envoy-upstream-service-time", "9045"},
teslamate_1  |       {"x-frame-options", "DENY"},
teslamate_1  |       {"x-content-type-options", "nosniff"},
teslamate_1  |       {"strict-transport-security", "max-age=31536000; includeSubDomains"},
teslamate_1  |       {"cache-control", "no-cache, no-store, private, s-max-age=0"},
teslamate_1  |       {"date", "Fri, 27 Oct 2023 01:57:41 GMT"},
teslamate_1  |       {"server", "envoy"}
teslamate_1  |     ],
teslamate_1  |     body: %{
teslamate_1  |       "error" => "{\"error\": \"timeout\"}",
teslamate_1  |       "error_description" => "",
teslamate_1  |       "response" => nil
teslamate_1  |     },
teslamate_1  |     status: 408,
teslamate_1  |     opts: [
teslamate_1  |       access_token: "stripped"
teslamate_1  |     ],
teslamate_1  |     __module__: TeslaApi,
teslamate_1  |     __client__: %Tesla.Client{fun: nil, pre: [], post: [], adapter: nil}
teslamate_1  |   }
teslamate_1  | }
teslamate_1  |     (teslamate 1.28.0-dev) lib/tesla_api/vehicle.ex:67: TeslaApi.Vehicle.get_with_state/2
teslamate_1  |     (teslamate 1.28.0-dev) lib/teslamate/api.ex:47: TeslaMate.Api.get_vehicle_with_state/2
teslamate_1  |     (teslamate 1.28.0-dev) lib/teslamate/vehicles/vehicle.ex:1212: TeslaMate.Vehicles.Vehicle.fetch_with_reachable_assumption/2
teslamate_1  |     (elixir 1.15.6) lib/task/supervised.ex:101: Task.Supervised.invoke_mfa/2
teslamate_1  |     (elixir 1.15.6) lib/task/supervised.ex:36: Task.Supervised.reply/4
teslamate_1  | Function: #Function<1.121950876/0 in TeslaMate.Vehicles.Vehicle.handle_event/4>
teslamate_1  |     Args: []
```

**Comment by brontide** [comment]
Specifically pull #3373 seems to update overview data as expected and may be timing out because I'm not driving.  Will confirm on the morning.

**Comment by brontide** [comment]
I'm now pretty confident this is the same as #3371 with pull #3373 being the fix since my drive this morning was captured.  I will go ahead and close this as a duplicate with instructions ( likely to be obsolete once #3373 is merged ).

```
git clone https://github.com/jsight/teslamate.git
cd teslamate
git checkout patched

```

Once complete you will need to update your docker-compose to point to that directory rather than the current image

```
services:
  teslamate:
    #image: teslamate/teslamate:edge
    build: temp/teslamate  # this is the path to the git
```

Then run

```
docker-compose build
docker-compose up -d
```

**Comment by Tazmanian79** [comment]
Will there follow an update of Teslamate that include this patch?
I have a docker container running but not really into these commands, only used to update by pulling a new image ;-)

**Comment by cwanja** [comment]
> Will there follow an update of Teslamate that include this patch?
> 
> I have a docker container running but not really into these commands, only used to update by pulling a new image ;-)

That is really up to @adriankumpf. Not clear at the moment.

**Comment by merlinfive** [comment]
Perfect ! , got update to 2023.38.4 this morning and your explanations works like charm again👍 
2023 S Plaid

![image](https://github.com/adriankumpf/teslamate/assets/101570702/adc796f0-ad46-40ad-bb96-c303a621f680)

**Comment by JakobLichterfeld** [comment]
related PR: #3373

**Comment by kinteseay** [comment]
Hey, I followed the instructions and thought I had things up and running, as I saw both of my vehicles in the teslamate interface.  However, after driving my model 3 for a drive, the data showed up on my Model S info in Grafana.  It's as if my car IDs aren't correct.  My model S was 1 and the Model 3 was 2.  OR 0 and 1 (not sure). Any idea what I can do?

**Comment by bikeymouse** [comment]
Mmm, I build the docker container as per instructions, and it seems to work, the car show up in the dashboard in again. However with my first drives this morning it only recorded this partly. 

Also in the logs is see the error " (KeyError) key :drive_state not found in: %TeslaApi.Error", which occurs every minute the car is online:
`
2023-10-28T09:06:48.842424982Z 2023-10-28 09:06:48.842 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/xxxremovedxxx/vehicle_data -> 408 (9084.751 ms)
2023-10-28T09:06:48.843275605Z 2023-10-28 09:06:48.842 [error] Task #PID<0.16445.0> started from :"1" terminating
2023-10-28T09:06:48.843287912Z ** (KeyError) key :drive_state not found in: %TeslaApi.Error{
2023-10-28T09:06:48.843291918Z   reason: :unknown,
2023-10-28T09:06:48.843294636Z   message: "%{\"error\" => \"{\\\"error\\\": \\\"timeout\\\"}\", \"error_description\" => \"\", \"response\" => nil}",
2023-10-28T09:06:48.843299368Z   env: %Tesla.Env{
2023-10-28T09:06:48.843302175Z     method: :get,
2023-10-28T09:06:48.843304743Z     url: "https://owner-api.teslamotors.com/api/1/vehicles/xxxremovedxxx/vehicle_data",
2023-10-28T09:06:48.843326716Z     query: [],
2023-10-28T09:06:48.843330609Z     headers: [
2023-10-28T09:06:48.843333599Z       {"x-xss-protection", "1; mode=block"},
2023-10-28T09:06:48.843336321Z       {"x-txid", "xxxremovedxxx"},
2023-10-28T09:06:48.843339101Z       {"content-type", "application/json; charset=utf-8"},
2023-10-28T09:06:48.843342024Z       {"x-request-id", "xxxremovedxxx"},
2023-10-28T09:06:48.843344664Z       {"content-length", "75"},
2023-10-28T09:06:48.843347517Z       {"x-envoy-upstream-service-time", "9051"},
2023-10-28T09:06:48.843350116Z       {"x-frame-options", "DENY"},
2023-10-28T09:06:48.843352505Z       {"x-content-type-options", "nosniff"},
2023-10-28T09:06:48.843354908Z       {"strict-transport-security", "max-age=31536000; includeSubDomains"},
2023-10-28T09:06:48.843357483Z       {"cache-control", "no-cache, no-store, private, s-max-age=0"},
2023-10-28T09:06:48.843360291Z       {"date", "Sat, 28 Oct 2023 09:06:48 GMT"},
2023-10-28T09:06:48.843363204Z       {"server", "envoy"}
2023-10-28T09:06:48.843365991Z     ],
2023-10-28T09:06:48.843368476Z     body: %{
2023-10-28T09:06:48.843371047Z       "error" => "{\"error\": \"timeout\"}",
2023-10-28T09:06:48.843373887Z       "error_description" => "",
2023-10-28T09:06:48.843376632Z       "response" => nil
2023-10-28T09:06:48.843379066Z     },
2023-10-28T09:06:48.843381442Z     status: 408,
2023-10-28T09:06:48.843383918Z     opts: [
2023-10-28T09:06:48.843387349Z       access_token: "xxxx removed xxx"
2023-10-28T09:06:48.843391887Z     ],
2023-10-28T09:06:48.843394554Z     __module__: TeslaApi,
2023-10-28T09:06:48.843397292Z     __client__: %Tesla.Client{fun: nil, pre: [], post: [], adapter: nil}
2023-10-28T09:06:48.843399878Z   }
2023-10-28T09:06:48.843402172Z }
2023-10-28T09:06:48.843404682Z     (teslamate 1.28.0-dev) lib/tesla_api/vehicle.ex:59: TeslaApi.Vehicle.get_with_state/2
2023-10-28T09:06:48.843412811Z     (teslamate 1.28.0-dev) lib/teslamate/api.ex:47: TeslaMate.Api.get_vehicle_with_state/2
2023-10-28T09:06:48.843415987Z     (elixir 1.15.6) lib/task/supervised.ex:101: Task.Supervised.invoke_mfa/2
2023-10-28T09:06:48.843418545Z     (elixir 1.15.6) lib/task/supervised.ex:36: Task.Supervised.reply/4
2023-10-28T09:06:48.843421055Z Function: #Function<1.121950876/0 in TeslaMate.Vehicles.Vehicle.handle_event/4>
2023-10-28T09:06:48.843424097Z     Args: []
`

**Comment by brontide** [comment]
@bikeymouse you have not missed anything, that is an attempt by the software to capture the driving state.  If you are not driving you will get that error.  I had this error and the patch worked despite the above errors when parked.

**Comment by marcbaier** [comment]
> Will there follow an update of Teslamate that include this patch? I have a docker container running but not really into these commands, only used to update by pulling a new image ;-)

I would really like that too. I have no idea how to manually install these patches / updates on my Synology NAS / Docker installation of Teslamate. So far there is no update of the teslamate:latest version.

**Comment by sparvata** [comment]
Can you tell me where this needs to be cloned into? The folder where the docker-compose.yml file is present? If not then how does the services change know where to find the built image? Sorry for my ignorance, I am not very familiar with the docker configurations.
> > ```
> > git clone https://github.com/jsight/teslamate.git
> > cd teslamate
> > git checkout patched
> > ```

services:
  teslamate:
    #image: teslamate/teslamate:edge
    build: temp/teslamate  # this is the path to the git

**Comment by merlinfive** [comment]
yesterday I got 2023.38.6 and it seems something was changed again

I got I guess all informations but Firmeware is not listed anymore

![image](https://github.com/adriankumpf/teslamate/assets/101570702/2c870a8a-d871-4a07-9f1c-cbac473d639f)
![image](https://github.com/adriankumpf/teslamate/assets/101570702/443056fe-b8e6-499c-be72-22349d2dc519)

**Comment by cwanja** [comment]
> yesterday I got 2023.38.6 and it seems something was changed again
> 
> 
> 
> I got I guess all informations but Firmeware is not listed anymore
> 
> 
> 
> ![image](https://github.com/adriankumpf/teslamate/assets/101570702/2c870a8a-d871-4a07-9f1c-cbac473d639f)
> 
> ![image](https://github.com/adriankumpf/teslamate/assets/101570702/443056fe-b8e6-499c-be72-22349d2dc519)
> 
> 

Do your drives data log and TeslaMate interface work?

**Comment by merlinfive** [comment]
yes , drives seems ok , there are entrys after the update

![image](https://github.com/adriankumpf/teslamate/assets/101570702/e282a11e-81f4-4d63-a694-59d19a039b21)

and also state changes today a few times

![image](https://github.com/adriankumpf/teslamate/assets/101570702/d0cbb5e9-fb27-451c-a137-7e5bbb9e96b3)


seems only FW number is missing to not problematic at the moment

**Comment by Runescom** [comment]
I get the following error: 
ERROR: yaml.scanner.ScannerError: mapping values are not allowed here
  in "./docker-compose.yml", line 8, column 15

version: "3"

services:
  teslamate:
   #image: teslamate/teslamate:latest
   build: /home/ubuntu/teslamate/teslamate
restart: always
    depends_on:
      - database

Can someone see the issue?

**Comment by Tovrin** [comment]
When will the standard docker image be updated?  Unfortunately, I don't have git on my machine so I can't patch the version as mentioned above.

**Comment by JakobLichterfeld** [comment]
> When will the standard docker image be updated?  Unfortunately, I don't have git on my machine so I can't patch the version as mentioned above.

That's why I created a fork which includes the patch and build the docker images automatically for all platforms (more discussion in the other PR)
 https://github.com/teslamate-fork/teslamate-fork#installation-of-the-fork-with-the-fix-for-tesla-fw-2023381

**Comment by sentinalll** [comment]
After I ran docker compose with custom build image Tesla Model Y had already twice opened frunk without my notice and Tesla started alarming because of that. Did someone have the same issue?
Firmware version: 2023.38.6
 I did what was suggested below:
> git clone https://github.com/jsight/teslamate.git
cd teslamate
git checkout patched

**Comment by JakobLichterfeld** [comment]
As TeslaMate does not have an API send function, this has nothing to do with Teslamate, and yes, that's intentional.

**Comment by JakobLichterfeld** [comment]
As TeslaMate does not have an API send function, this has nothing to do with Teslamate, and yes, that's intentional.

This issue is closed, as Teslamate 1.27.4 solved the issue.

**Comment by sentinalll** [comment]
Thanks for the reply. Anyway, I disabled teslamate for now and will see if the issue will appear again. In my case, the frunk opened just when I navigated between dashboards. Will try to rub the latest docker image available and check the behaviorr.
